backend/module/chaos/serverexperimentation/xds/xds_cache.go

```diff
@@ -105,8 +130,14 @@ func setSnapshot(resource []gcpTypes.ResourceWithTtl, cluster string, snapshotCa
 		return nil
 	}
 
-	logger.Infow("Setting snapshot", "cluster", cluster, "resource", resource)
-	snapshot := gcpCacheV3.NewSnapshotWithTtls(computedVersion, nil, nil, nil, nil, resource, nil)
+	snapshot := gcpCacheV3.Snapshot{}
+	logger.Infow("Setting snapshot", "cluster", cluster, "resource", resource, "isECDSResource", isECDSResource)
+	if isECDSResource {
```

Comment:
maybe just pass through the type name? or even pass `map[string][]gcpTypes.ResourceWithTtl` to make this a bit more generic